src/pkg/p2p/preheat/models/policy/policy.go

```diff
@@ -118,6 +118,19 @@ type Trigger struct {
 	} `json:"trigger_setting,omitempty"`
 }
 
+// ValidatePreheatPolicy validate preheat policy
+func (s *Schema) ValidatePreheatPolicy() error {
+	// currently only validate cron string of preheat policy
+	if s.Trigger != nil && s.Trigger.Type == TriggerTypeScheduled && len(s.Trigger.Settings.Cron) > 0 {
```

Comment:
duplicate validation in the length of cron string

Reply:
This is because we need to skip `utils.ValidateCronString()` validation for cron with empty string "". In preheat and tag retention, it uses empty string cron to unschedule an existing job.